e2etests/test/pnp_telemetry.js

```diff
@@ -0,0 +1,178 @@
+        return new Promise(function (resolve) {
+          setTimeout(resolve, 3000);
+        });
+      }).then(function () {
+        deviceClient.open(function (openErr) {
+          rdv.imIn('deviceClient');
+          if (openErr) {
+            done(openErr);
+          } else {
+            deviceClient.sendTelemetry({fake: 'payload'}, componentName, function (sendErr) {
+              if (sendErr) {
+                done(sendErr);
+              }
+              debug('trying to finish from the sending side');
+              rdv.imDone('deviceClient');
+            });
+          }
+        });
+      })
+      .catch(function (err) {
+        debug(`Error thrown by Event Hubs client: ${err}`);
+        done(err);
+      });
+    });
+
+    it('sends a telemetry message without a component name and the service sees the message with the correct component name, content type, and content encoding', function (done) {
+      const rdv =  new Rendezvous(done);
+      const startAfterTime = Date.now() - 5000;
+
+      const onEventHubMessage = function (eventData) {
+        if (eventData.annotations['iothub-connection-device-id'] === provisionedDevice.deviceId) {
+          if (eventData.properties.content_type !== 'application/json') {
+            debug(`eventData.properties.content_type "${eventData.properties.content_type}" doesn't match "application/json"`);
+          } else if (eventData.properties.content_encoding !== 'utf-8') {
+            debug(`eventData.properties.content_encoding "${eventData.properties.content_encoding}" doesn't match "utf-8"`);
+          } else if (eventData.annotations['dt-subject']) {
+            debug(`eventData.annotations['dt-subject'] "${eventData.annotations['dt-subject']}" should be undefined`);
+          } else if (JSON.stringify(eventData.body) !== JSON.stringify({fake: 'payload'})) {
+            debug(`eventData.body "${JSON.stringify(eventData.body)}" doesn't match "${JSON.stringify({fake: 'payload'})}"`);
+          } else {
+            debug('trying to finish from the receiving side');
+            rdv.imDone('ehClient');
+          }
+        } else {
+          debug(`Incoming device id is: ${eventData.annotations['iothub-connection-device-id']}`);
+        }
+      };
+
+      const onEventHubError = function (err) {
+        debug(`Error from Event Hub Client Receiver: ${err}`);
+        done(err);
+      };
+
+      EventHubClient.createFromIotHubConnectionString(hubConnectionString).then(function (client) {
+        ehClient = client;
+        rdv.imIn('ehClient');
+        return ehClient.getPartitionIds();
+      }).then(function (partitionIds) {
+        partitionIds.forEach(function (partitionId) {
+          ehClient.receive(partitionId, onEventHubMessage, onEventHubError, { eventPosition: EventPosition.fromEnqueuedTime(startAfterTime) });
+        });
+        return new Promise(function (resolve) {
+          setTimeout(resolve, 3000);
+        });
+      }).then(function () {
```

Comment:
So.  Much.  Copy.  Paste.  (say it in William Shatner's voice).  

(not really your fault because this is a well-worn pattern in these E2E tests long before you got here, but it's a habit we need to stop.)

Reply:
Refactored the test to use `EventHubReceiverHelper` to be cleaner and avoid repeating code.